### DISCUSSION TITLE
Hybrid storage model

### DISCUSSION THREAD

**Discussion by SleepingShell** [rook/rook#6803] [Q&A] [answered]
I was wondering if there is a native way to create a hybrid storage pool with rook-ceph natively (using the CRDs and not having to edit the cluster with ``rook-ceph-tools``).

I am wanting to achieve something similar to: https://www.root314.com/ceph/2017/04/30/Ceph-hybrid-storage-tiers/

More specifically, I have a single-node cluster (although this shouldn't matter for the solution) in which I have the following:
2xSSDs
4xHDDs
1xNVMe

I would like to create a ``CephBlockPool`` with 2 replicas - one replica is always on an SSD and one is always on a HDD. This would allow me to utilize the full storage capacity of the SSDs and retain their read speed, as opposed to only having 50% of the raw SSD capacity usable.

**Comment by travisn** [comment, marked as answer]
@SleepingShell There is not a way currently to have a hybrid storage pool unless you use the toolbox as mentioned. The pool `deviceClass` setting is the closest feature, but it will put all replicas on the same media type. Want to open an feature for this? Thanks!

**Reply by SleepingShell** [reply]
Thanks for the quick reply @travisn !

https://github.com/rook/rook/issues/6805

**Reply by BlaineEXE** [reply]
It's also worth noting that Ceph will block on writes until all writes have finished, so your write speed will be limited to HDD speed. Reads should still be able to happen at SSD speed, however.

**Reply by SleepingShell** [reply]
Yup, that is the goal :) Is there any estimate on when this will be included?
